virtual_rainforest/models/abiotic/energy_balance.py

```diff
@@ -0,0 +1,321 @@
+    ) -> None:
+        """Initializes point-based energy_balance method.
+
+        Generates a set of climate, conductivity, and soil parameters for running the
+        first time step of the model. This might go in the json file.
+        """
+
+        # set boundary conditions
+        self.canopy_layers = canopy_layers
+        self.soil_layers = soil_layers
+        self.canopy_height = canopy_height
+        self.soil_depth = soil_depth
+
+        self.air_temperature_2m = data["air_temperature_2m"]
+        self.relative_humidity_2m = data["relative_humidity_2m"]
+        self.atmospheric_pressure_2m = data["atmospheric_pressure_2m"]
+        self.wind_speed_10m = data["wind_speed_10m"]
+        self.mean_annual_temperature = mean_annual_temperature
+        self.soil_moisture_reference = data["soil_moisture_reference"]
+
+        # interpolate initial temperature profile, could be more realistic
+        self.temperature_above_canopy = self.air_temperature_2m
+        temperature_interpolation = np.linspace(
+            self.mean_annual_temperature,
+            self.temperature_above_canopy,
+            self.canopy_layers + self.soil_layers,
+        )
+
+        self.air_temperature = temperature_interpolation[
+            (int(self.soil_layers)) : int((self.canopy_layers + self.soil_layers))
+        ]
+        self.soil_temperature = temperature_interpolation[0 : int(self.soil_layers)][
+            ::-1
+        ]
+        self.absorbed_radiation = absorbed_radiation
+        self.canopy_temperature = (
+            self.air_temperature + LEAF_TEMPERATURE_INI_FACTOR * self.absorbed_radiation
+        )
+
+        # initiate relative humidity and atmospheric pressure
+        self.relative_humidity = np.repeat(
+            data["relative_humidity_2m"], self.canopy_layers
+        )
+        self.atmospheric_pressure = data["atmospheric_pressure_2m"]
+
+        # set initial conductivities
+        self.air_conductivity = (
+            np.repeat(AIR_CONDUCTIVITY, self.canopy_layers + 1)
+            * (self.canopy_layers / self.canopy_height)
+            * (2 / self.canopy_layers)
+        )
+        self.air_conductivity[0] = 2 * self.air_conductivity[0]
+        self.air_conductivity[self.canopy_layers] = (
+            self.air_conductivity[self.canopy_layers]
+            * (self.canopy_height / self.canopy_layers)
+            * 0.5
+        )
+
+        self.leaf_conductivity = np.linspace(
+            np.array(MIN_LEAF_CONDUCTIVITY),
+            np.array(MAX_LEAF_CONDUCTIVITY),
+            self.canopy_layers,
+        )
+        self.air_leaf_conductivity = np.linspace(
+            np.array(MIN_LEAF_AIR_CONDUCTIVITY),
+            np.array(MAX_LEAF_AIR_CONDUCTIVITY),
+            self.canopy_layers,
+        )
+
+        # set initial heights
+        self.canopy_node_heights = [
+            (x + 0.5) / self.canopy_layers * self.canopy_height
+            for x in range(0, self.canopy_layers)
+        ]
```

Comment:
Two things here:

1. Just checking: Is the order of operations here definitely correct? You don't mean `(x + 0.5) / (self.canopy_layers * self.canopy_height)`?
2. This could be written more simply using numpy:

```py
self.canopy_node_heights = (np.arange(self.canopy_layers) + 0.5) / self.canopy_layers * self.canopy_height
```